### PR TITLE
twitter reason add #798

### LINKED ISSUE #798: feat: add reasons to the marker detail (tweet)
## Define your request
Add reason badges to twitter card

-- fill in the line below:
** discord username: orhanozkercin#0001 **

## Extra context
image

### PR DESCRIPTION
**Pull request by OrhanOzkercin**
# Description

add reasons to twitter card

**discord username: orhanozkercin#0001**

**closes #798**

Please describe your changes. Also describe your aim and content. Do not forget to list the dependencies required caused by those changes.

## Things to check before creating a PR
- [ ] I have inspected my topic, checked.
- [ ] If it is a core feature, executed detailed tests.

## Creating PR rules
- [ ] PR must be created for an issue with approved tag. Otherwise PR will be rejected.
- [ ] Relevant issue number: The issue number related to PR must be attached to head of PR header, after prefix after prefix # must be attached in parenthesis. A header like this could be used "prefix(#issue_number): PR header" .
- [ ] A descriptive and understandable title: The PR title should clearly describe the nature and purpose of the changes. The PR title should be the first thing displayed when the PR is opened. And it should follow the semantic commit rules. For example, a title like "docs(#issueId): Add README.md" can be used.
- [ ] Related file selection: Only relevant files should be touched and no other files should be affected.
- [ ] Format and Lint Suitability : The code should be made in accordance with a certain format standard and examined according to the lint rules.
- [ ] Clean commit history: The commits where the changes are made should be clear and organized.
- [ ] Opening PR when job is completed: PR should be opened when job is completed and sent for review by other team members.

## Changes

- [ ] A new feature (a change that adds a new feature, not a breaking change)
- [ ] A new refactor (a change that is not a breaking change, that improves the readability or performance of the code)
- [ ] A breaking change
- [ ] Documentation change

# How were these changes tested?

Please describe the tests you did to test the changes you made. Please also specify your test configuration.

**Test Configuration**:

* Firmware version:
* Hardware:
* Toolchain:
* SDK:

### CODE REVIEW

File: components/UI/Drawer/components/channels/twitter/PlaceholderTweet.tsx [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
 type Props = {
   source: TwitterParameters;
+  reason: string;
   full_text?: FeedChannelTwitterProps["full_text"];
 };
 
-const PlaceholderTweet = ({ source, full_text }: Props) => {
+const PlaceholderTweet = ({ source, reason, full_text }: Props) => {
+  const isNotNan = (value: any) => {
+    return !!value && value !== "NaN" && value !== "nan";
```
Comment (cemiluzunhasan) [inline components/UI/Drawer/components/channels/twitter/PlaceholderTweet.tsx:14]: Very small enhancement, we can convert last 2 conditions to
`value.toString().toLowerCase() !== "nan"`

If the PR is urgent, please ignore the comment. Good job!

Comment (OrhanOzkercin) [reply components/UI/Drawer/components/channels/twitter/PlaceholderTweet.tsx]: Done! @cemiluzunhasan

**Review by cemiluzunhasan — DISMISSED** [review]

**Review by cemiluzunhasan — DISMISSED** [review]

File: components/UI/Drawer/components/channels/babala/FeedChannelBabala.tsx [hunk 2]
```diff
@@ -2,20 +2,17 @@ import Typography from "@mui/material/Typography";
 import Chip from "@mui/material/Chip";
 import { FeedChannelBabalaProps } from "../../types";
 import { Divider } from "@mui/material";
+import { capitalize, isNaN } from "@/utils/helpers";
```
Comment (usirin) [inline components/UI/Drawer/components/channels/babala/FeedChannelBabala.tsx:5]: 💯

File: components/UI/Drawer/components/channels/babala/FeedChannelBabala.tsx [hunk 3]
```diff
@@ -29,15 +26,16 @@ const FeedChannelBabala = ({
...
         </div>
         <Divider />
         <Typography style={styles.fullText}>{full_text}</Typography>
         {extraValues}
+
+        <div style={styles.chip_container}>
+          {!isNaN(reason) && reason && (
+            <Chip label={capitalize(reason)} color="info" />
+          )}
+        </div>
```
Comment (usirin) [inline components/UI/Drawer/components/channels/babala/FeedChannelBabala.tsx:38]: i think we should wrap the outer element with this conditional as well

File: components/UI/Drawer/components/channels/twitter/EmbedTweet.tsx [hunk 4]
```diff
@@ -1,14 +1,15 @@
-import { useEffect, useState } from "react";
+import React, { useEffect, useState } from "react";
```
Comment (usirin) [inline components/UI/Drawer/components/channels/twitter/EmbedTweet.tsx:1]: No need to add `React` import

**Review by usirin — DISMISSED** [review]

**Review by EmirBoyaci — DISMISSED** [review]

File: components/UI/Drawer/components/channels/babala/FeedChannelBabala.tsx (hunk 3 above)
Comment (OrhanOzkercin) [reply components/UI/Drawer/components/channels/babala/FeedChannelBabala.tsx:38]: You are right, I'll fix it!

File: components/UI/Drawer/components/channels/twitter/EmbedTweet.tsx (hunk 4 above)
Comment (OrhanOzkercin) [reply components/UI/Drawer/components/channels/twitter/EmbedTweet.tsx:1]: Yeap

**Comment by OrhanOzkercin** [comment]
@usirin @EmirBoyaci I updated the PR with your comments, could you approve again

**Review by EmirBoyaci — APPROVED** [review]

**Review by usirin — APPROVED** [review]

**Pull request merged**